app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class BaseController < ActionController::API
+      include CanCan::ControllerAdditions
+      include RequestTimeouts
+      include Pagy::Backend
+      include JsonApiPagination
+
+      RESULTS_PER_PAGE = 50
+
+      check_authorization
+
+      attr_accessor :current_api_user
+
+      before_action :authenticate_user
+
+      rescue_from Exception, with: :error_during_processing
+      rescue_from CanCan::AccessDenied, with: :unauthorized
+      rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
+      private
+
+      def authenticate_user
+        return if (@current_api_user = request.env['warden'].user)
+
+        if api_key.blank?
+          # An anonymous user
+          @current_api_user = Spree::User.new
+          return
+        end
+
+        return if (@current_api_user = Spree::User.find_by(spree_api_key: api_key.to_s))
+
+        invalid_api_key
+      end
```

Comment:
Can we remove assignments from conditionals? I would find this a lot easier to read:
```suggestion
      def authenticate_user
        @current_api_user = warden_user || api_key_user || guest_user
        invalid_api_key if @current_api_user.nil?
      end
      
      def warden_user
        request.env['warden'].user
      end

      def api_key_user
        Spree::User.find_by(spree_api_key: api_key.to_s)
      end
      
      def guest_user
         Spree::User.new if api_key.blank?
      end
```

Reply:
Ah, nice. I pulled this method straight from our existing API base controller and didn't re-assess it. That looks much better! :heart:

Reply:
nice refactor @mkllnk 👏 
In the new version, is authenticate_user returning @current_api_user if @current_api_user is not nil? I'd write a unit test to make it clear.
Or maybe the last line could be written as a ternary op to make it explicit?
`@current_api_user.present? ? @current_api_user : invalid_api_key`

Reply:
nice refactor @mkllnk 👏 
In the new version, is authenticate_user returning @current_api_user if @current_api_user is not nil? I'd write a unit test to make it clear.
Or maybe the last line could be written as a ternary op to make it explicit?
`@current_api_user.present? ? @current_api_user : invalid_api_key`